File: go/test/protocol/test_case_action_send_telemetry.go
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
package protocol

import (
	"gopkg.in/yaml.v3"
)

type testCaseActionSendTelemetry struct {
	TelemetryName  *string             `yaml:"telemetry-name"`
	TopicTokenMap  map[string]string   `yaml:"topic-token-map"`
	Timeout        *TestCaseDuration   `yaml:"timeout"`
	TelemetryValue *string             `yaml:"telemetry-value"`
	Metadata       *map[string]string  `yaml:"metadata"`
	CloudEvent     *TestCaseCloudEvent `yaml:"cloud-event"`
	Qos            *int                `yaml:"qos"`
}

type TestCaseActionSendTelemetry struct {
	testCaseActionSendTelemetry
}

func (sendTelemetry *TestCaseActionSendTelemetry) UnmarshalYAML(
	node *yaml.Node,
) error {
	*sendTelemetry = TestCaseActionSendTelemetry{}

	sendTelemetry.TelemetryName = TestCaseDefaultInfo.Actions.SendTelemetry.GetTelemetryName()
	sendTelemetry.TelemetryValue = TestCaseDefaultInfo.Actions.SendTelemetry.GetTelemetryValue()
	sendTelemetry.Qos = TestCaseDefaultInfo.Actions.SendTelemetry.GetQos()

	err := node.Decode(&sendTelemetry.testCaseActionSendTelemetry)

	if sendTelemetry.Timeout == nil {
		defaultTimeout := TestCaseDefaultInfo.Actions.SendTelemetry.GetTimeout()
		sendTelemetry.Timeout = &defaultTimeout
	}

	return err
}

File: dotnet/test/Azure.Iot.Operations.Protocol.MetlTests/TestCaseActionSendTelemetry.cs
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

namespace Azure.Iot.Operations.Protocol.MetlTests
{
    public class TestCaseActionSendTelemetry : TestCaseAction
    {
        public static string? DefaultTelemetryName;
        public static string? DefaultTelemetryValue;
        public static TestCaseDuration? DefaultTimeout;
        public static int? DefaultQos;

        public string? TelemetryName { get; set; } = DefaultTelemetryName;

        public Dictionary<string, string>? TopicTokenMap { get; set; }

        public TestCaseDuration? Timeout { get; set; } = DefaultTimeout;

        public string? TelemetryValue { get; set; } = DefaultTelemetryValue;

        public Dictionary<string, string>? Metadata { get; set; }

        public TestCaseCloudEvent? CloudEvent { get; set; }

        public int? Qos { get; set; } = DefaultQos;
    }
}
